es/blog_index.go

```diff
@@ -59,10 +59,7 @@ func (index *BlogIndex) ReindexAll() error {
 func (index *BlogIndex) Update(scope Scope) error {
 	log.Debugf("BlogIndex.Update - Scope: %+v.", scope)
 	removed, err := index.removeFromIndex(scope)
-	if err != nil {
-		return err
-	}
-	return index.addToIndex(scope, removed)
+	return utils.JoinErrors(err, index.addToIndex(scope, removed))
```

Comment:
readability issue, func args are computed regardless of err. While this is what we want, it's a little confusing.